net/client_grpc.go

```diff
@@ -311,9 +320,17 @@ func (g *grpcClient) conn(p Peer) (*grpc.ClientConn, error) {
 				metrics.GroupDialFailures.WithLabelValues(p.Address()).Inc()
 			}
 		}
-		g.conns[p.Address()] = c
-		metrics.GroupConnections.Set(float64(len(g.conns)))
+		if err != nil {
+			g.conns[p.Address()] = c
+		}
 	}
+
+	// Emit the connection state regardless of whether it's a new or an existing connection
+	if err != nil {
```

Comment:
I don't understand why you check if there was an error before firing the metric?
Why not fire it all the time?

Reply:
and yes, you're right above. Fixing the comparison.